### PR TITLE
15 react stripe

### PR DESCRIPTION
**Pull request by rkerins91**
To use this locally, you'll need to create a Stripe account and get the keys from your dashboard. You need to add your stripe secret key to .env in server/ as STRIPE_SECRET_KEY and you need to replace my publishable key on line 292 of ReactStripe.js to your publishable key.

### CODE REVIEW

File: client/src/pages/Balance/ReactStripe.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    });
+
+    const confirm = await stripe.confirmCardPayment(
+      `${requestPaymentIntent.data.clientSecret}`,
+      {
+        payment_method: payload.paymentMethod.id
+      }
+    );
+
+    if (confirm.paymentIntent.status === "succeeded") {
```
Comment (ashan36) [inline client/src/pages/Balance/ReactStripe.js]: When payment fails, confirm.paymentIntent is undefined and the app crashes.
Tested with a couple of the card numbers https://stripe.com/docs/testing#cards that are meant to get a specific error response.

File: client/src/pages/Balance/ReactStripe.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        fill="#6772e5"
+        d="M8.5,7.29791847 L6.12604076,4.92395924 C5.79409512,4.59201359 5.25590488,4.59201359 4.92395924,4.92395924 C4.59201359,5.25590488 4.59201359,5.79409512 4.92395924,6.12604076 L7.29791847,8.5 L4.92395924,10.8739592 C4.59201359,11.2059049 4.59201359,11.7440951 4.92395924,12.0760408 C5.25590488,12.4079864 5.79409512,12.4079864 6.12604076,12.0760408 L8.5,9.70208153 L10.8739592,12.0760408 C11.2059049,12.4079864 11.7440951,12.4079864 12.0760408,12.0760408 C12.4079864,11.7440951 12.4079864,11.2059049 12.0760408,10.8739592 L9.70208153,8.5 L12.0760408,6.12604076 C12.4079864,5.79409512 12.4079864,5.25590488 12.0760408,4.92395924 C11.7440951,4.59201359 11.2059049,4.59201359 10.8739592,4.92395924 L8.5,7.29791847 L8.5,7.29791847 Z"
+      />
+    </svg>
+    {children}
+  </div>
+);
+
+const ResetButton = ({ onClick }) => (
+  <button type="button" className="ResetButton" onClick={onClick}>
```
Comment (ashan36) [inline client/src/pages/Balance/ReactStripe.js]: Where are the SubmitButton or ResetButton components defined here used? I can't find it.

File: client/src/pages/Balance/ReactStripe.js [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+          autoComplete="name"
+          value={billingDetails.name}
+          onChange={e => {
+            setBillingDetails({ ...billingDetails, name: e.target.value });
+          }}
+        />
+      </fieldset>
+      <fieldset className="FormGroup">
+        <CardField
+          onChange={e => {
```
Comment (ashan36) [inline client/src/pages/Balance/ReactStripe.js]: This CardField and the name Field above it are display at different heights. Should probably be the same. Not a big deal, but it was perceptible from afar.

File: server/routes/user.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ const { check, body, validationResult } = require("express-validator");
 const router = express.Router();
 const bcrypt = require("bcryptjs");
 const config = require("../config/config");
+const stripe = require("stripe")(process.env.STRIPE_SECRET_KEY);
```
Comment (ashan36) [inline server/routes/user.js]: Can you add this env variable to the server config file?

**Review by ashan36 — CHANGES_REQUESTED** [review]
This branch is also 53 commits behind dev. There are some weird styling happening with the background of the login and signup pages.

The checkout dialog looks really nicely done.

**Review by nicholasrobinharrison — COMMENTED** [review]
Probably need to pull dev into this branch to make sure nothing is going to break

File: client/src/pages/Balance/ReactStripe.js (hunk 3 above)
Comment (rkerins91) [reply client/src/pages/Balance/ReactStripe.js]: I am not seeing this issue on my machine. Not sure what you mean.

**Review by ashan36 — APPROVED** [review]

**Review by nicholasrobinharrison — APPROVED** [review]

**Pull request merged**